src/Raven.Server/Documents/Sharding/Queries/ShardedQueryProcessorBase.cs

```diff
@@ -39,7 +39,11 @@ public abstract class ShardedQueryProcessorBase<TCombinedResult> : AbstractShard
     protected void ApplyPaging(ref TCombinedResult result, QueryTimingsScope scope)
     {
         QueryTimingsScope pagingScope = null;
-
+        Query.Offset ??= Query.Start;
+        Query.Limit ??= Query.PageSize;
```

Comment:
Instead of modifying the original `Query` we we'll be better off by adding local variables `offset`, `limit` and so something like: `var offset = Query.Offset ?? Query.Start ?? 0;` and use them in this function instead of checking `Query.Offset`